Cython/Includes/cpython/contextvars.pxd

```diff
@@ -0,0 +1,93 @@
+    # `obj` must not be NULL. This function always succeeds.
+
+    bint PyContextVar_CheckExact(object obj)
+    # Return `true` if `obj` is of type `PyContextVar_Type`.
+    # `obj` must not be NULL. This function always succeeds.
+
+    bint PyContextToken_CheckExact(object obj)
+    # Return `true` if `obj` is of type `PyContextToken_Type`.
+    # `obj` must not be NULL. This function always succeeds.
+
+    object PyContext_New()
+    # Return value: New reference.
+    # Create a new empty context object.
+    # Returns NULL if an error has occurred.
+
+    object PyContext_Copy(object ctx)
+    # Return value: New reference.
+    # Create a shallow copy of the passed `ctx` context object.
+    # Returns NULL if an error has occurred.
+
+    object PyContext_CopyCurrent()
+    # Return value: New reference.
+    # Create a shallow copy of the current thread context.
+    # Returns NULL if an error has occurred.
+
+    int PyContext_Enter(object ctx) except -1
+    # Set `ctx` as the current context for the current thread.
+    # Returns 0 on success, and -1 on error.
+
+    int PyContext_Exit(object ctx) except -1
+    # Deactivate the `ctx` context and restore the previous context
+    # as the current context for the current thread.
+    # Returns 0 on success, and -1 on error.
+
+    object PyContextVar_New(const char* name, object default_value)
+    # Return value: New reference.
+    # Create a new ContextVar object. The `name` parameter is used
+    # for introspection and debug purposes. The `default_value` parameter
+    # may optionally specify the default value for the context variable.
+    # If an error has occurred, this function returns NULL.
+
+    int PyContextVar_Get1 "PyContextVar_Get" (PyObject* var, PyObject* default_value, PyObject** value) except -1
+    # Get the value of a context variable.
+    # Returns -1 if an error has occurred during lookup, and 0 if no error
+    # occurred, whether or not a value was found.
+    #
+    # If the context variable was found, `value` will be a pointer to it.
+    # If the context variable was not found, `value` will point to:
+    #
+    #   • `default_value`, if not NULL;
+    #   • the default value of `var`, if not NULL;
+    #   • NULL
+    int PyContextVar_Get2 "PyContextVar_Get" (object var, object default_value, PyObject** value) except -1
+    # a different declaration of PyContextVar_Get that requires default values
+    # be passed on call.
+
+    object PyContextVar_Set(object var, object value)
+    # Return value: New reference.
+    # Set the value of `var` to `value` in the current context.
+    # Returns a pointer to a PyObject object, or NULL if an error has occurred.
+
+    int PyContextVar_Reset(object var, object token) except -1
+    # Reset the state of the `var` context variable to that it was in
+    # before `PyContextVar_Set()` that returned `token` was called.
+    # This function returns 0 on success and -1 on error.
```

Comment:
It would be nice to also add convenience functions like (obviously with tests):
```suggestion
    # This function returns 0 on success and -1 on error.


cdef inline object get_value(var, default_value=None):
    """Return a new reference to the value of the context variable,
    or the provided default value if no such value was found.
    """
    cdef PyObject *value
    PyContextVar_Get(var, <PyObject*>default_value, &value)
    return <object>value


cdef inline object get_value_or_none(var):
    """Return a new reference to the value of the context variable,
    or the default value of the context variable,
    or None if no such value or default was found.
    """
    cdef PyObject *value
    PyContextVar_Get(var, NULL, &value)
    return <object>value if value is not NULL else None
```

Reply:
(this addition can be a separate PR if you don't feel like writing tests for this)

Reply:
I added similar helper functions in ff16389c0da5acc2be1b49fd074b0919cb252b88..